app/main.py

```diff
@@ -1,14 +1,95 @@
-class Validator:
-    pass
+from typing import Any, Type
+from abc import ABC, abstractmethod
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(
+            self,
+            owner:
+            Type,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
 
+    def __get__(
+            self,
+            instance: object,
+            owner: Type
+    ) -> str:
+        return getattr(instance, self.protected_name)
 
-class OneOf:
-    pass
+    def __set__(
+            self,
+            instance: object,
+            value: Type
+    ) -> None:
+        self.validate(value)
+        setattr(instance, self.protected_name, value)
+
+    @abstractmethod
+    def validate(
+            self,
+            value: Any
+    ) -> None:
+        pass
+
+
+class Number(Validator):
+    def __init__(
+            self,
+            min_value: int,
+            max_value: int
+    ) -> None:
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def validate(
+            self,
+            value: Any
+    ) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Quantity should be integer.")
+        if value < self.min_value or value > self.max_value:
+            raise ValueError(f"Quantity should not be less than "
+                             f"{self.min_value} and greater "
+                             f"than {self.max_value}.")
+
+
+class OneOf(Validator):
+    def __init__(
+            self,
+            options: Any
+    ) -> None:
+        self.options = options
+
+    def validate(
+            self,
+            value: Any
+    ) -> None:
+        if value not in self.options:
+            raise ValueError(f"Expected {self.protected_name} to be one of {self.options}.")
```

Comment:
You should call `value` instead of `self.protected_name`:
```suggestion
            raise ValueError(f"Expected {value} to be one of {self.options}.")
```